gocourse05/ZoomTask/main.go

```diff
@@ -0,0 +1,58 @@
+/*
+Треба написати програму для умовної  обробки зображення різного типу, з використанням поліморфізму,
+контрактів і panic/recover
+*/
+package main
+
+import "fmt"
+
+type ImageProcessor interface {
+	Process() error
+}
+
+type BMPImageProcessor struct {
+	path string
+}
+
+type PNGImageProcessor struct {
+	path string
+}
+
+func (i *PNGImageProcessor) Process() error {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Printf("We processed a panic, result = %v. Image: %s\n", r, i.path)
+		} else {
+			fmt.Println("Everything finished successfully")
+		}
+	}()
+
+	fmt.Println("Trying to process png image")
+
+	var s *string
+	fmt.Println(*s)
+	return nil
+}
+
+func (i *BMPImageProcessor) Process() error {
+	fmt.Printf("We processed bmp image: %s\n", i.path)
+	return nil
+}
+
+func main() {
+	var bmp = BMPImageProcessor{
+		path: "./image/image.bmp",
+	}
+	var png = PNGImageProcessor{
```

Comment:
Так простіше:
```suggestion
	bmp := BMPImageProcessor{
		path: "./image/image.bmp",
	}
	png := PNGImageProcessor{
```

Знову ж таки, ось це можна відформатувати автоматично за допомогою `gofumpt`. Писав про це в Slack https://golang-beginning-6.slack.com/archives/C07EVP8S4FK/p1724071944955179

---
Не забувайте видаляти непотрібні пусті рядки, як ось тут:
```
func sum(a, b int) int {

	return a + b
}
```
`gofmt` таке пропускає, а я - ні.  В реальних проектах на десятки тисяч рядків, зайвих рядків може бути тисячу, а це вже немало. Наприклад, ось мій PR, де видалялись непотрібні рядки, його без проблем прийняли.

Щоб полегшити собі життя, варто поставити `gofumpt` https://github.com/mvdan/gofumpt. Цей тул є надбудовою над gofmt: видаляє непотрібні пусті рядки, сортує імпорти тощо. В README вказано точніше, що програма вміє робити.

Поставити такою командою:
```
go install mvdan.cc/gofumpt@latest
```

Щоб відформатувати всі файли:
```
gofumpt -w .
```

Також можна запускати ось так, якщо з якоїсь причини не працює попередня команда:
```
go run mvdan.cc/gofumpt@latest -w .
```

Reply:
не розумію чому в мене не працює після заміни ноута, буду розбиратись